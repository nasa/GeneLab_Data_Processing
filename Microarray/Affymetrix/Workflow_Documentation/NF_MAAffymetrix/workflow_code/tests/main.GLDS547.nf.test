nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "core"

    test("GLDS-547: Should run without failures") {

        when {
            params {
                // define parameters here. Example: 
                osdAccession = "OSD-547"
                gldsAccession = "GLDS-547"
                // biomart_attribute = "AFFY HTA 2 0" IN ISA
                limit_biomart_query = 300
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                            path("${ params.outputDir }/Metadata").list(),
                            path("${ params.outputDir }/00-RawData").list(),
                            path("${ params.outputDir }/01-oligo_NormExp").list(),
                            path("${ params.outputDir }/02-limma_DGE").list(),
                            ).match()
        }

    }

}
