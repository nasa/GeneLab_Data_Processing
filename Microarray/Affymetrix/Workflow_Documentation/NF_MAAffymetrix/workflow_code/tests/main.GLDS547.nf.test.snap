{
    "GLDS-547: Should run without failures": {
        "content": [
            [
                "OSD-547_metadata_OSD-547-ISA.zip:md5,417d4b50fae769c0c42f1b3589f145e6",
                "OSD-547_microarray_v0_runsheet.csv:md5,46f96ff3d2069330aa7589ebd4239545"
            ],
            [
                "raw_intensities_probe.csv:md5,c70428bfa9f23ebd6093761391e00ae8"
            ],
            [
                "normalized_expression_probeset.csv:md5,fa9d6eb92180457d81fdc1ce42e684c1",
                "normalized_intensities_probe.csv:md5,f11890e076445706868512a7abba301f"
            ],
            [
                "SampleTable.csv:md5,5b8a41b5f38582b553344f7a7d075ff5",
                "contrasts.csv:md5,cdf56af4c7ec6fe6692a2da22ecb0ef6",
                "differential_expression.csv:md5,41208f1164a6b8fd84006bde8776cd8c",
                "visualization_PCA_table.csv:md5,87efc5e6591c24a15d36153b354aca1c"
            ]
        ],
        "timestamp": "2023-05-18T17:12:20+0000"
    }
}